alchemy-web/docs/providers/supabase/function.md

```diff
@@ -0,0 +1,220 @@
+      );
+      
+      const { data } = await supabase.from('users').select('*');
+      return new Response(JSON.stringify(data));
+    });
+  `,
+});
+```
+
+### Function with Adoption
+
+```typescript
+// This will adopt an existing function if one with the same name already exists
+const existingFunction = Function("existing-function", {
+  projectRef: "my-project-ref",
+  adopt: true,
+  body: `
+    export default function handler(req: Request) {
+      return new Response("Updated function");
+    }
+  `,
+});
+```
+
+### Function that Won't be Deleted
+
+```typescript
+const persistentFunction = Function("persistent-function", {
+  projectRef: "my-project-ref",
+  delete: false, // Function will not be deleted when resource is destroyed
+  body: `
+    export default function handler(req: Request) {
+      return new Response("This function persists");
+    }
+  `,
+});
+```
+
+## API Operations
+
+### Create Function
+- **Endpoint**: `POST /projects/{projectRef}/functions`
+- **Body**: Function configuration including slug, name, body, import_map, etc.
+- **Response**: Function object with ID and initial status
+
+### Get Function
+- **Endpoint**: `GET /projects/{projectRef}/functions/{slug}`
+- **Response**: Full function details including current version
+
+### Deploy Function
+- **Endpoint**: `POST /projects/{projectRef}/functions/{slug}/deploy`
+- **Body**: Updated function code and configuration
+- **Response**: 200 on successful deployment
+
+### Delete Function
+- **Endpoint**: `DELETE /projects/{projectRef}/functions/{slug}`
+- **Response**: 200 on successful deletion
+
+### List Functions
+- **Endpoint**: `GET /projects/{projectRef}/functions`
+- **Response**: Array of function objects
+
+## Error Handling
+
+The resource handles the following error scenarios:
+
+- **409 Conflict**: When `adopt: true` is set, the resource will attempt to find and adopt an existing function with the same name
+- **Rate Limiting**: Automatic exponential backoff for 429 responses
+- **Server Errors**: Automatic retry for 5xx responses
+- **404 on Delete**: Ignored (function already deleted)
+
+## Lifecycle Management
+
+- **Creation**: Functions are created with the specified code and configuration
+- **Updates**: When function body is provided during updates, the function is automatically deployed with the new code
+- **Deletion**: Functions can be deleted unless `delete: false` is specified
+
+## Dependencies
+
+Functions depend on:
+- **Project**: Must specify a valid `projectRef`
+
+## Runtime Environment
+
+Edge Functions run in a Deno runtime with the following characteristics:
+
+- **Runtime**: Deno with Web APIs
+- **Timeout**: 150 seconds maximum execution time
+- **Memory**: 512MB maximum memory usage
+- **Cold Start**: Functions may experience cold starts after periods of inactivity
+- **Environment Variables**: Access to project environment variables via `Deno.env.get()`
+
+## Security Considerations
```

Comment:
None of the docs should have this. Please understand the documentation style of the repo and update all the docs you wrote